ethminer/MinerAux.h

```diff
@@ -92,6 +93,47 @@ class MinerCLI
 		Stratum
 	};
 
+	MinerCLI() : 
+		m_io_work(m_io_service), 
+		m_io_work_timer(m_io_service),
+		m_io_strand(m_io_service)
+	{
+		// Post first deadline timer to give io_service
+		// initial work
+		m_io_work_timer.expires_from_now(boost::posix_time::seconds(60));
```

Comment:
Why this fake job is needed?

Reply:
Every time io_service starts allocates it's own thread_pool.
When there are no more jobs in the poll queue it destroys all it's threads and recreation is very long time consuming. Also on Windows apparently it does not restart properly.
This fake job gives the poll queue always something to do so it never stops.